### PR TITLE
conditionals-programs_with_decision_points-hard-q7

### PR DESCRIPTION
**Pull request by ketphan02**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [ ]  Are the assertion fail messages good?
    - [ ]  Logical (does it make sense?)
    - [ ]  Readable (can a student understand what’s wrong?)
    - [ ]  Not too feed-y (does it give away the answer?)
    - [ ]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [ ]  Specific (does the fail message relate to the question?)
- [ ]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [ ]  EOF newlines
- [ ]  Do the comments for Parsons follow the correct convention?
- [ ]  Is the question being tested sufficiently?
    - [ ]  Are there representative cases to test the program?
    - [ ]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [ ]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [ ] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/conditionals/programs_with_decision_points/hard/q7/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    RiskyBusiness.main(new String[0]);
+  }
+
+  static Stream<Arguments> inputProviderValid() {
+    return Stream.of(
+        Arguments.of(1, "Low Risk"),
+        Arguments.of(100, "Low Risk"),
+        Arguments.of(999, "Low Risk"),
+        Arguments.of(1000, "Moderate Risk"),
+        Arguments.of(5000, "Moderate Risk"),
```
Comment (Hedgemon4) [inline src/conditionals/programs_with_decision_points/hard/q7/MainTest.java:39]: I would try to not always test the exact middle number, something more random is less predictable (maybe 6739 for example).

File: src/conditionals/programs_with_decision_points/hard/q7/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    TestOption.incorrectStructureErrorMessage = "Your program does not print the correct risk level with the given valid input.";
+    runWithInput(population + "", new Clause[] {
+        new StringLiteral(riskLevel)
+    });
+  }
+
+  @ParameterizedTest
+  @MethodSource("inputProviderInvalid")
+  public void printsErrorMessageWithInvalidInput(int population) throws InvalidClauseException {
+    TestOption.incorrectStructureErrorMessage = "Your program does not print the error message for the invalid input.";
```
Comment (Hedgemon4) [inline src/conditionals/programs_with_decision_points/hard/q7/MainTest.java:64]: Remove the in `... error message for the invalid input.`

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/conditionals/programs_with_decision_points/hard/q7/RiskyBusiness.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package conditionals.programs_with_decision_points.hard.q7;
+
+/*
+ * Write a program that asks the user to enter the number of people present in a certain area, and display the
+ * approximate risk of infection based on the following information: Less than 1000 people is "Low Risk", Between
```
[the remark this answers is not in the document]
Comment (ketphan02) [reply src/conditionals/programs_with_decision_points/hard/q7/RiskyBusiness.java:5]: @Hedgemon4 I don't think I understand this

[the remark this answers is not in the document]
Comment (Hedgemon4) [reply src/conditionals/programs_with_decision_points/hard/q7/RiskyBusiness.java:5]: Oh, I just think that it is random to just be like `display the approximate risk of infection` I think it should be something like `display the approximate risk of contracting the flu` or something

[the remark this answers is not in the document]
Comment (Hedgemon4) [reply src/conditionals/programs_with_decision_points/hard/q7/RiskyBusiness.java:5]: It just felt a little clunky to me

File: src/conditionals/programs_with_decision_points/hard/q7/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    TestOption.incorrectStructureErrorMessage = "Your program does not print the correct risk level with the given valid input.";
+    runWithInput(population + "", new Clause[] {
+        new StringLiteral(riskLevel)
+    });
+  }
+
+  @ParameterizedTest
+  @MethodSource("inputProviderInvalid")
+  public void printsErrorMessageWithInvalidInput(int population) throws InvalidClauseException {
+    TestOption.incorrectStructureErrorMessage = "Your program does not print the error message for invalid input.";
```
Comment (Hedgemon4) [inline src/conditionals/programs_with_decision_points/hard/q7/MainTest.java:64]: `an error message`

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/conditionals/programs_with_decision_points/hard/q7/RiskyBusiness.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package conditionals.programs_with_decision_points.hard.q7;
+
+/*
+ * Write a program that asks the user to enter the number of people present in a certain area, and display the
+ * approximate risk of disease transmission based on the following information: Less than 1000 people is "Low Risk",
+ * Between 1000 and 10000 people is "Moderate Risk", above 10000 people is "High Risk." Make sure to display an error
+ * message if the user provides an invalid input.
```
Comment (Hedgemon4) [inline src/conditionals/programs_with_decision_points/hard/q7/RiskyBusiness.java:7]:
```suggestion
 * message if the user provides an invalid input (numbers less than or equal to 0).
```

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]

**Pull request merged**